docs/firmware/golioth-firmware-sdk/light-db-stream/guide-lightdb-stream-type.md

```diff
@@ -0,0 +1,142 @@
+---
+id: by-type
+title: LightDB Stream (by type)
+sidebar_position: 1
+---
+
+Type-based LightDB functions can be used to send a single time-series data value
+to the cloud. A timestamp is automatically added to the data by the server and
+stored using a specific path that you provide. This data will accumulate along
+with all previously received values. If the chosen path does not already exist
+it will be created.
+
+Synchronous and asynchronous functions exist to handle several data types:
+`bool`, `float`, `int`, `string`.
+
+## Includes
+
+```c
+#include "golioth.h"
+```
+
+Including the `golioth.h` header file makes the Golioth API functions available
+to your program.
+
+## Synchronous
+
+```c
+int err;
+bool my_bool = true;
+float my_float = 1.337;
+int my_int = 42;
+char *my_str[] = "Golioth";
+
+// Stream a bool value
+err = golioth_lightdb_stream_set_bool_sync(client, "sensor/active", my_bool, 1);
+
+// Stream a float value
+err = golioth_lightdb_stream_set_float_sync(client, "sensor/reading", my_float, 1);
+
+// Stream an int value
+err = golioth_lightdb_set_int_sync(client, "sensor/calib", my_int, 1);
+
+// Stream a string
+err = golioth_lightdb_stream_set_string_sync(client,
+                                             "sensor/name",
+                                             my_str,
+                                             strlen(my_str),
+                                             1);
+```
+
+The simplest way to send LightDB Stream data to Golioth is using a synchronous
+function call. This blocking function will send data to the given path and wait
+for a response (or error) from the server. If a response is not received within
+the given timeout, a `GOLIOTH_ERR_TIMEOUT` error code will be returned.
+
+In the above examples, we tell Golioth to set the `"sensor/active"` path to the
+value of a variable called `my_bool`, with a `1` second timeout. When Golioth
+receives this data, the server will automatically add a timestamp based on when
+it was received. Functions for three other data types are also demonstrated,
+accomplishing the same end goal of sending individual time-series data points to
+Golioth.
+
+:::tip
+
+Each function call represents a separate network payload being sent from the
+device to Golioth. If you have several data points that need to be updated at
+the same time, you can use a JSON or CBOR object to do so. For the example
+above, an object containing all values could be sent to the `sensors` path,
+turning four function calls into just one.
```

Comment:
```suggestion
turning four function calls (and four network requests) into just one.
```

I think we should re-emphasize that the benefit here is fewer network interactions.